### PR TITLE
Added Ape Keys Popup

### PR DESCRIPTION
**Pull request by Miodec**
Added popup for generating, enabling/disabling, renaming and deleting ape keys

### CODE REVIEW

File: frontend/src/scripts/db.ts [hunk 1]
```diff
@@ -58,12 +59,13 @@ export async function initSnapshot(): Promise<
...
     // }
     // LoadingPage.updateText("Downloading user...");
-    const [userData, configData, tagsData, presetsData] = (
+    const [userData, configData, tagsData, presetsData, apeKeysData] = (
       await Promise.all([
         Ape.users.getData(),
         Ape.configs.get(),
         Ape.users.getTags(),
         Ape.presets.get(),
+        Ape.apeKeys.get(),
```
Comment (Bruception) [inline frontend/src/scripts/db.ts:68]: Why load these on every refresh?

File: frontend/src/scripts/popups/ape-keys-popup.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import * as DB from "../db";
+import Ape from "../ape";
+import * as Loader from "../elements/loader";
+import * as Notifications from "../elements/notifications";
+
+function refreshList(): void {
+  const data = DB.getSnapshot().apeKeys;
+  const table = $("#apeKeysPopupWrapper table tbody");
+  table.empty();
+  if (Object.keys(data).length === 0) {
```
Comment (Bruception) [inline frontend/src/scripts/popups/ape-keys-popup.ts:10]: Refactor,

`const apeKeyIds = Object.keys(data)`

File: frontend/src/scripts/popups/ape-keys-popup.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    table.append(
+      "<tr><td colspan='6' style='text-align: center;'>No keys found</td></tr>"
+    );
+    return;
+  }
+  for (const keyId of Object.keys(data)) {
```
Comment (Bruception) [inline frontend/src/scripts/popups/ape-keys-popup.ts:16]: ```apeKeyIds.forEach```

File: frontend/src/scripts/types/types.d.ts [hunk 4]
```diff
@@ -241,6 +241,15 @@ declare namespace MonkeyTypes {
     hash?: string;
   }
 
+  interface ApeKeys {
+    [key: string]: {
+      name: string;
+      enabled: boolean;
+      createdOn: number;
+      modifiedOn: number;
```
Comment (Bruception) [inline frontend/src/scripts/types/types.d.ts:249]: Needs to be updated.

File: frontend/src/scripts/types/types.d.ts [hunk 5]
```diff
@@ -403,6 +412,7 @@ declare namespace MonkeyTypes {
     quoteMod?: boolean;
     discordId?: string;
     config?: Config;
+    apeKeys: ApeKeys;
```
Comment (Bruception) [inline frontend/src/scripts/types/types.d.ts:415]: Why is this part of the config?

Comment (Bruception) [inline frontend/src/scripts/types/types.d.ts:415]: Needs to be optional.

File: frontend/src/scripts/db.ts (hunk 1 above)
Comment (Miodec) [reply frontend/src/scripts/db.ts:68]: Removed

File: frontend/src/scripts/popups/ape-keys-popup.ts [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+              key.enabled
+                ? `<i class="fas fa-check-square"></i>`
+                : `<i class="far fa-fw fa-square"></i>`
+            }
+          </div>
+        </td>
+        <td>${key.name}</td>
+        <td>${moment(key.createdOn).format("DD MMM YYYY HH:mm")}</td>
+        <td>${moment(key.modifiedOn).format("DD MMM YYYY HH:mm")}</td>
+        <td>-</td>
```
Comment (Bruception) [inline frontend/src/scripts/popups/ape-keys-popup.ts:34]: We should consume the new field for ape keys.

**Comment by Bruception** [comment]
LGTM IIW

**Review by Bruception — APPROVED** [review]

**Comment by Ferotiq** [comment]
I'll do a quick review

**Comment by Ferotiq** [comment]
Looks good to me, too.

**Pull request merged**